File: Jazz.web.frame/net/Jazz.Helper.DataBase/Common/IDataBase.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Data.Common;
using System.Data;

namespace Jazz.Helper.DataBase.Common
{
    public interface IDataBase
    {
        DbConnection getConnection();

        #region ExecuteNonQuery命令

        int ExecuteNonQuery(string safeSql, DbTransaction tran);

        int ExecuteNonQuery(string sql, DbTransaction tran, params DbParameter[] values);

        #endregion

        #region ExecuteScalar命令
        int ExecuteScalar(string safeSql);

        int ExecuteScalar(string sql,params DbParameter[] values);

        #endregion

        #region ExecuteReader命令

        DbDataReader ExecuteReader(string safeSql, DbConnection Connection);

        DbDataReader ExecuteReader(string sql, DbConnection Connection, params DbParameter[] values);

        #endregion

        #region ExecuteDataTable命令

          DataTable ExecuteDataTable(CommandType type, string safeSql, params DbParameter[] values);

          DataTable ExecuteDataTable(string safeSql);

          DataTable ExecuteDataTable(string sql, params DbParameter[] values);

        #endregion

        #region GetDataSet命令

          DataSet GetDataSet(string safeSql, string tabName, params DbParameter[] values);

        #endregion

        #region ExecureData命令

          int ExecureData(DataSet ds, string strTblName);

          int ExecureData(DataSet ds, string SQL, string strTblName);

          int ExecureData(DataSet ds, string SQL, string strTblName, DbParameter[] pars);

          void BulkToDB(DataTable dt, string tbName);

        #endregion

    }


}

File: Jazz.web.frame/java/jazz.helper.database/src/org/jazz/helper/database/common/IDataBase.java
package org.jazz.helper.database.common;

import java.sql.Connection;
import java.sql.SQLException;
import java.util.HashMap;
import java.util.List;

public interface IDataBase {
	
	/**
	 * 获得数据库连接
	 * @return
	 */
	Connection getConnection();
	
	/**
	 * 开始事务
	 * @return {@code Connection} 数据库连接
	 */
	Connection beginTran();
	
	/**
	 * 回滚事务
	 * @param con
	 * @throws SQLException
	 */
	void rollbackTran(Connection con) throws SQLException;
	
	/**
	 * 提交事务
	 * @param con
	 * @return
	 */
	int commintTran(Connection con);
	
	/**
	 * 执行数据库数据修改语句
	 * @param safeSql 
	 * @param conn 启用事务事务时传入连接字符串
	 * @return
	 */
	int ExecuteNonQuery(String safeSql,Connection conn);
	
	/**
	 * 执行数据库修改语句
	 * @param safeSql
	 * @param params 
	 * @param conn 启用事务事务时传入连接字符串
	 * @return
	 */
	int ExecuteNonQuery(String safeSql,HashMap<String,Object> params,Connection conn);
	
	/**
	 * 获得查询图
	 * @param safeSql
	 * @return {@code List<HashMap<String,Object>>}
	 */
	List<HashMap<String,Object>> ExecuteDataTable(String safeSql);
	
	/**
	 * 获得查询图
	 * @param Sql
	 * @param params
	 * @return {@code List<HashMap<String,Object>>}
	 */
	List<HashMap<String,Object>> ExecuteDataTable(String Sql,HashMap<String,Object> params);
	
	/**
	 * 获得查询图
	 * @param sql sql语句
	 * @param params 参数图
	 * @param cls 实体类
	 * @return {@code List<HashMap<String,Object>>}
	 */
	<T> List<T> ExecuteDataTable(String sql,HashMap<String,Object> params,Class<T> cls);
	
	/**
	 * 获得统计值
	 * @param safeSql
	 * @return
	 */
	int ExecuteScalar(String safeSql);

	/**
	 * 获得统计值
	 * @param sql
	 * @param params
	 * @return
	 */
    int ExecuteScalar(String sql,HashMap<String,Object> params);

	
}
